### PR TITLE
Refactor process controller scripts

### PR DESCRIPTION
**Pull request by novicecpp**
Fix https://github.com/dmwm/CRABServer/issues/8436

Final version of https://github.com/dmwm/CRABServer/pull/8589, also fix https://github.com/dmwm/CRABServer/issues/8539 and https://github.com/dmwm/CRABServer/issues/8496 .

A lot of code changed (mostly shell scripts) but it should be less complicated and have more structure.

Please remind me if I forget anything from the last time we had chat.

### CODE REVIEW

File: cicd/crabserver_pypi/run.sh [hunk 1]
```diff
@@ -14,9 +14,9 @@ mkfifo /data/srv/state/crabserver/crabserver-fifo
 # to pipe. It is safe because only single process can read from pipe at the time
 cat /data/srv/state/crabserver/crabserver-fifo &
 
-#start the service
-#export CRYPTOGRAPHY_ALLOW_OPENSSL_102=true
-/data/start.sh -c
+# start the service
+# manage.py required service name, anything is fine because we never use it.
```
Comment (belforte) [inline cicd/crabserver_pypi/run.sh:18]: why do you keep this argument if we never use it ?
(N.B. "required" or "requires" ? past or present ?)

File: cicd/crabserver_pypi/manage_skel.sh
Comment (belforte) [inline cicd/crabserver_pypi/manage_skel.sh:1]: am I correct that this is never used ? Why is here then ? Sort of "documentation" ?

File: cicd/crabtaskworker_pypi/Publisher/manage.sh [hunk 2]
```diff
@@ -1,34 +1,19 @@
...
 
-# some variable use in start_srv
-CONFIG="${SCRIPT_DIR}"/current/PublisherConfig.py
-
-helpFunction() {
-    grep "^##H" "${0}" | sed -r "s/##H(| )//g"
-}
+# sanity check
+if [[ -z ${COMMAND+x} || -z ${MODE+x} || -z ${DEBUG+x} || -z ${SERVICE+x} ]]; then
+    >&2 echo "All envvars are not set!."
```
Comment (belforte) [inline cicd/crabtaskworker_pypi/Publisher/manage.sh:14]: English: Not all envvars are set

**Review by belforte — APPROVED** [review]

**Comment by belforte** [comment]
thanks @novicecpp and sorry for very late review.
I like the new structure. I have only added a few minor comments.

File: cicd/crabserver_pypi/manage_skel.sh
Comment (novicecpp) [reply cicd/crabserver_pypi/manage_skel.sh:1]: It never use. I can mov it to doc, the first one in "Bits and pieces" :)

File: cicd/crabtaskworker_pypi/Publisher/manage.sh (hunk 2 above)
Comment (novicecpp) [reply cicd/crabtaskworker_pypi/Publisher/manage.sh:14]: Thanks.

File: cicd/crabserver_pypi/run.sh (hunk 1 above)
Comment (novicecpp) [reply cicd/crabserver_pypi/run.sh:18]: Ah sorry. I misunderstand the question.
The SERVICE arg is only use in publisher. It is already setup by runContainer.sh script in TW but not the rest.  
At the time I dev manage.py, make it as require arguments is much simpler to code.

But let me try again to make it optional. It might be easier than I thought.

File: cicd/crabserver_pypi/manage_skel.sh
Comment (novicecpp) [reply cicd/crabserver_pypi/manage_skel.sh:1]: I do not think we need this anymore because it is unlikely we will have more services than we do now.

File: cicd/crabserver_pypi/run.sh (hunk 1 above)
Comment (novicecpp) [reply cicd/crabserver_pypi/run.sh:18]: Well, turn out it is easier than I thought :)

**Comment by novicecpp** [comment]
Thanks again Stefano for review.
